### DISCUSSION TITLE
Connection with remote development server

### DISCUSSION THREAD

**Discussion by dietercoopman** [spatie/ray#174] [Q&A]
Hi there , i have a question related to development on a remote server.  

Most of our projects are developed on a remote server ( via sftp ) .  Programming is done in PhpStorm with an upload on save deployment.  Is it possible to connect ray with this remote machine - in fact let the remote machine connect with my local machine ?  I guess i will need to juggle with the ip address of my internet connection and port forwarding ?  
Maybe there  is a more straight forward solution for this.   The juggling stuff would also be very difficult in a team approach ,because i would have to assign a port per developer to do the right forward per client.

Thanks for the great work you've done with Ray , i love it ( using it with Laravel Sail )

**Comment by smoetje** [comment]
All my projects are running a through a linux server on the central synology nas. I am also wondering how to let ray connect to the address of this central development server in the network, instead of localhost. Right now I can only configure the port in the application.

**Reply by dietercoopman** [reply]
Your situation is indeed similar , but my server is even outside my lan ... so my client is in a private network and the server on a public ( with no vpn possibilities to it 🤷🏼‍♂️ ).

**Comment by Terrorboy** [comment]
I also need it.
We have purchased it and are waiting for the feature to be supported.

**Comment by patinthehat** [comment]
@Terrorboy @dietercoopman @smoetje 
Until a proper solution is implemented, have you tried something like [ngrok](https://ngrok.com/docs) as a workaround?

**Reply by Terrorboy** [reply]
Before official support, ngrok will work.
But it is too cumbersome.

![image](https://user-images.githubusercontent.com/5427199/106992541-04625d80-67bc-11eb-8acc-69b0889db68d.png)



ps. Apart from the above, there may be cases where you need to have multiple hosts. Are you considering it?

**Reply by Terrorboy** [reply]
See: https://github.com/spatie/ray/discussions/119

**Reply by dietercoopman** [reply]
Haven't tried it ( and will not 😂 ) because the tool needs to handle this ( as not being oss ).  I've seen on Twitter the spatie team is working on it ... we all know that team likes some challenge 👍 

https://twitter.com/freekmurze/status/1356284627648991240?s=21

**Reply by dietercoopman** [reply]
But I also think ngrok ( or expose from Marcel pociot ) could be a solution ... if the multi client on the same public ip can be handled with it

**Reply by dietercoopman** [reply]
> See: https://github.com/spatie/ray/discussions/119

Oh , cool hadn't seen the reply from Freek

**Reply by marcorieser** [reply]
I adapted @Terrorboy's approach to use [expose from beyond code](https://beyondco.de/docs/expose/introduction):
* Install expose and run this in the terminal
`expose share http://localhost:23517`
* Insert the generated remote host and port 80 into the ray config
```php
return [
    ...
    'host' => 'xyz.sharedwithexpose.com',
    'port' => 80,
    ...
];
```

**Reply by dietercoopman** [reply]
This is now natively implemented via ssh , what a great job !🍾
